File: app/src/main/java/com/softwareoverflow/hiitultimate/database/dao/BaseDao.java
package com.softwareoverflow.hiitultimate.database.dao;

import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;

import java.util.List;


interface BaseDao<T> {

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    long createOrUpdate(T toCreate);

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void createOrUpdate(List<T> toCreate);

    @Delete
    int delete(T toDelete);
}

File: app/src/main/java/com/softwareoverflow/hiit_trainer/data/dao/BaseDao.kt
package com.softwareoverflow.hiit_trainer.data.dao

import androidx.room.Dao
import androidx.room.Insert
import androidx.room.OnConflictStrategy

@Dao
interface BaseDao<T> {

    @Insert (onConflict = OnConflictStrategy.REPLACE)
    suspend fun createOrUpdate(obj: T) : Long

    fun delete(obj: T)

}